client/mysqldump.c

```diff
@@ -787,22 +787,34 @@ static void write_header(FILE *sql_file, char *db_name)
             "/*!40111 SET @OLD_SQL_NOTES=@@SQL_NOTES, SQL_NOTES=0 */;\n",
             path?"":"NO_AUTO_VALUE_ON_ZERO",compatible_mode_normal_str[0]==0?"":",",
             compatible_mode_normal_str);
+
+    // This is specifically to allow MyRocks to bulk load a dump faster
+    // We have no interest in anything earlier than 5.7 and 17 being the
+    // current release. 5.7.8 and after can only use P_S for session_variables
+    // and never I_S. So we first check that P_S is present and the
+    // session_variables table exists. If no, we simply skip the optimization
+    // assuming that MyRocks isn't present either. If it is, ohh well, bulk
+    // loader will not be invoked.
     fprintf(sql_file,
-            "/*!50112 SELECT COUNT(*) INTO @is_rocksdb_supported FROM"
-            " INFORMATION_SCHEMA.SESSION_VARIABLES WHERE"
-            " VARIABLE_NAME='rocksdb_bulk_load' */;\n"
-            "/*!50112 SET @save_old_rocksdb_bulk_load ="
-            " IF (@is_rocksdb_supported,"
-            " 'SET @old_rocksdb_bulk_load = @@rocksdb_bulk_load',"
-            " 'SET @dummy_old_rocksdb_bulk_load = 0') */;\n"
-            "/*!50112 PREPARE s FROM @save_old_rocksdb_bulk_load */;\n"
-            "/*!50112 EXECUTE s */;\n"
-            "/*!50112 SET @enable_bulk_load = IF (@is_rocksdb_supported,"
-            " 'SET SESSION rocksdb_bulk_load = 1',"
-            " 'SET @dummy_rocksdb_bulk_load = 0') */;\n"
-            "/*!50112 PREPARE s FROM @enable_bulk_load */;\n"
-            "/*!50112 EXECUTE s */;\n"
-            "/*!50112 DEALLOCATE PREPARE s */;\n");
+            "/*!50717 SET @rocksdb_bulk_load_var_name='rocksdb_bulk_load' */;\n"
+            "/*!50717 SELECT COUNT(*) INTO @rocksdb_has_p_s_session_variables"
+            " FROM INFORMATION_SCHEMA.TABLES WHERE TABLE_SCHEMA LIKE"
+            " 'PERFORMANCE_SCHEMA' AND TABLE_NAME LIKE 'SESSION_VARIABLES'"
```

Comment:
Lowercase 'performance_schema', 'session_variables' - it should be more robust to match the schema name case too